UBCSailbot/sailbot_workspace
# Integration Test Framework

### Description

- Resolves #142 

#### Config Files
This is where we define our tests. 

`required_packages` is basically a list of stuff we want to have running for the test, and any extra config options/files we want to give it. Non-ROS executables like virtual_iridium and the website are to be put here too, they'll just be handled explicitly.

`inputs` is a list of inputs. I'm thinking the `type` can be something like ROS or HTTP, the `name` is the topic name or URL in the case of HTTP, and `body` is the actual input data. The `body` will need to be processed differently depending on the `type`, but we should be able to generically set key value pairs based on the `type`. 

`expected_outputs` is formatted the exact same way as the `inputs`, just on the other end.

#### Test Flow
1. Run the python script with a config file.
2. Start all specified packages.
3. Start monitoring output topics.
4. Send all inputs.
5. Over the span of 3(?) seconds, check that the expected outputs appear.



### Verification


### Resources

-
